cmd/auparse/auparse.go

```diff
@@ -163,52 +165,52 @@ func outputMultipleMessages(msgs []*auparse.AuditMessage) {
 
 	switch *format {
 	case "json":
-		if err := printJSON(event); err != nil {
+		if err := s.printJSON(event); err != nil {
 			log.WithError(err).Error("failed to marshal event to JSON")
 		}
 	case "yaml":
-		fmt.Println("---")
-		if err := printYAML(event); err != nil {
+		s.output.Write([]byte("---\n"))
+		if err := s.printYAML(event); err != nil {
 			log.WithError(err).Error("failed to marshal message to YAML")
 		}
 	default:
 		sm := event.Summary
-		fmt.Println("---")
-		fmt.Printf(`time="%v" sequence=%v category=%v type=%v actor=%v/%v action=%v thing=%v/%v how=%v tags=%v`+"\n",
+		s.output.Write([]byte("---\n"))
+		s.output.Write([]byte(fmt.Sprintf(`time="%v" sequence=%v category=%v type=%v actor=%v/%v action=%v thing=%v/%v how=%v tags=%v`+"\n",
 			event.Timestamp, event.Sequence, event.Category, event.Type, sm.Actor.Primary, sm.Actor.Secondary,
-			sm.Action, sm.Object.Primary, sm.Object.Secondary, sm.How, event.Tags)
+			sm.Action, sm.Object.Primary, sm.Object.Secondary, sm.How, event.Tags)))
 	}
 }
 
-func outputSingleMessage(m *auparse.AuditMessage) {
+func (s *streamHandler) outputSingleMessage(m *auparse.AuditMessage) {
 	switch *format {
 	case "json":
-		if err := printJSON(m.ToMapStr()); err != nil {
+		if err := s.printJSON(m.ToMapStr()); err != nil {
 			log.WithError(err).Error("failed to marshal message to JSON")
 		}
 	case "yaml":
-		if err := printYAML(m.ToMapStr()); err != nil {
+		if err := s.printYAML(m.ToMapStr()); err != nil {
 			log.WithError(err).Error("failed to marshal message to YAML")
 		}
 	default:
-		fmt.Printf("type=%v msg=%v\n", m.RecordType, m.RawData)
+		s.output.Write([]byte(fmt.Sprintf("type=%v msg=%v\n", m.RecordType, m.RawData)))
 	}
 }
 
-func printJSON(v interface{}) error {
+func (s *streamHandler) printJSON(v interface{}) error {
 	jsonBytes, err := json.Marshal(v)
 	if err != nil {
 		return err
 	}
-	fmt.Println(string(jsonBytes))
+	s.output.Write(append(jsonBytes, []byte("\n")...))
```

Comment:
This can be accomplished without the `append`, which causes an allocation, by using two separate `s.output.Write` calls.

The same applies to `printYAML`.